### ISSUE TITLE
`get-assignment` is wrong with CDCL solver

### ISSUE THREAD

**Issue by bclement-ocp** [OCamlPro/alt-ergo#937]
Consider the test `tests/models/bool/bool3.models.smt2` which we only test using the Tableaux solver (this is an issue, we should be testing model generation on the default solver now that this is supported).

```smt-lib
(set-logic ALL)
(set-option :produce-assignments true)
(set-option :produce-models true)
(declare-const x Bool)
(declare-const y Bool)
(assert (or (! (and x y) :named foo) (! (and (not x) (not y)) :named bar)))
(check-sat)
(get-model)
(get-assignment)
```

Alt-Ergo with the (default -- using `alt-ergo tests/models/bool/bool3.models.smt2`) CDCL solver finds the following model and assignment:
```smt-lib
unknown
(
  (define-fun x () Bool false)
  (define-fun y () Bool false)
)
((foo false)
 (bar false))
```

This is wrong: `bar` should be `true`, not `false` (`(and (not false) (not false))` is clearly `true`).

I believe this should be fixed by switching to the new Dolmen implementation of `:named` terms, since that will automatically declare `foo` and `bar` as actual constants.

**Comment by bclement-ocp** [comment]
> this is an issue, we should be testing model generation on the default solver now that this is supported

Before someone beats me to the punch: PR incoming for that.

**Comment by Stevendeo** [comment]
Can we close this issue?

**Comment by bclement-ocp** [comment]
Looks like this is fixed in #961 and we forgot to link it, yes.
